modules/openapi-generator/src/main/resources/python/python-experimental/model_templates/classvars.mustache

```diff
@@ -62,7 +62,7 @@
 
 {{#additionalPropertiesType}}
     @cached_property
-    def additional_properties_type():
+    def additional_properties_type(cls):
```

Comment:
Why did you add cls as an input here? The cls input is not used.
- this same question applies to openapi_types + discriminator

Reply:
@spacether all methods on a class technically require one argument. Which is self for instances of the object or the cls for a class method. The only exception is if it's decorated with `@staticmethod`. Although it "works" without this first argument mypy detects it (correctly) as an error.

Now that I look at this again for these it really should be called self not cls. This won't affect the functionality but I'll make the change anyway as it's confusing for it to have the wrong name.

Reply:
Actually ignore me. I misunderstood your question. I think this is a little more complicated to do correctly because of the way the @cached_property decorator works.

Reply:
@spacether So I've updated this to self so that `@ cached_property ` can be swapped for `@property` without any code changes.

Reply:
If we could avoid this error from being triggered in mypy that would be ideal. Without a cls or self input arg this code runs fine. Doesn't that make it valid python?